test/distributed/org/apache/cassandra/distributed/test/AdaptiveCompactionConfigTest.java

```diff
@@ -0,0 +1,140 @@
+                                              "'adaptive': 'true', " +
+                                              "'adaptive_interval_sec': 1, " +
+                                              "'adaptive_threshold': 0.01, " +
+                                              "'adaptive_min_cost': 1, " +
+                                              "'adaptive_starting_scaling_parameter': '0', " +
+                                              "'min_sstable_size_in_mb': '1'};"));
+            cluster.get(1).runOnInstance(() ->
+                                             {
+                                                 ColumnFamilyStore cfs = Keyspace.open("ks").getColumnFamilyStore("tbl");
+                                                 CompactionStrategy cs = cfs.getCompactionStrategy();
+                                                 Controller controller = cs.getController();
+                                                 assertTrue(controller instanceof AdaptiveController);
+                                                 int scalingParam = controller.getScalingParameter(0);
+                                                 //scaling parameter on L0 should be 0 to start
+                                                 assert scalingParam == 0;
+                                             });
+            /*
+                write and read some data.  logs should show that current scaling parameter configurations are written to file.
+                read/write costs are very small, so scaling parameters will not change
+             */
+            for (int i = 0; i < 10; i++)
+            {
+                for (int j = 0; j < 10; j++)
+                {
+                    createSSTables(cluster.get(1), "ks", "tbl", i, j);
+                    readData(cluster.get(1), "ks", "tbl");
+                }
+            }
+            cluster.get(1).runOnInstance(() ->
+                                         {
+                                             ColumnFamilyStore cfs = Keyspace.open("ks").getColumnFamilyStore("tbl");
+                                             CompactionStrategy cs = cfs.getCompactionStrategy();
+                                             Controller controller = cs.getController();
+                                             assertTrue(controller instanceof AdaptiveController);
+                                             /*
+                                                costs are not high enough to trigger a scaling parameter change, so
+                                                manually write new scaling parameters to see if it is picked up on restart
+                                              */
+                                             File directory = DatabaseDescriptor.getMetadataDirectory();
+                                             try
+                                             {
```

Comment:
There's a lot of duplicated code here and in `StoreScalingParameters()`. Can we have a single method that does most of this for us? e.g. by passing the scaling params as an argument?